### PR TITLE
Prototype HTML and CSS structure for the view with six-style icon fonts [#86021344]

### PR DESCRIPTION
**Pull request by jw-kaurand**
Prototype HTML and CSS structure for the view with six-style icon fonts.  Building LESS and including fonts via webpack.  Test is viewable at jwplayer/test/ui-test.html

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -36,6 +36,28 @@ module.exports = function(grunt) {
...
+                    noOverqualifying: false,
+                    noUnderscores: true,
+                    noUniversalSelectors: true,
+                    prefixWhitespace: true,
+                    strictPropertyOrder: true,
+                    zeroUnits: false,   // Occasionally set this to true, but it will misinterpret some values.
+                    includePaths: ['src/css', 'src/css/*']
+                },
+                files: {
+					'bin-debug/jwplayer.css' : 'src/css/jwplayer.less'
```
Comment (robwalch) [inline Gruntfile.js]: output to test/css-skin/

File: Gruntfile.js [hunk 2]
```diff
@@ -115,11 +143,26 @@ module.exports = function(grunt) {
                 ],
                 module: {
                     loaders: [
-                        {
-
-                            test: /\.less$/,
-                            loader: 'style-loader!css-loader!less-loader'
-                        }
+						{
```
Comment (robwalch) [inline Gruntfile.js]: Setup indentation and other style guide rules per https://github.com/jwplayer/jwplayer#style-guide

File: assets/fonts/svgs/audio-tracks.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg width="18" height="12" xmlns="http://www.w3.org/2000/svg"><g fill="none"><g fill="#000"><path d="M3 11.6v.4c-1.657 0-3-1.343-3-3 0-.44.095-.857.264-1.233C.094 7.207 0 6.615 0 6c0-3.314 2.686-6 6-6s6 2.686 6 6c0 .615-.093 1.208-.264 1.767.17.376.264.793.264 1.233 0 1.657-1.343 3-3 3V6c.744 0 1.424.27 1.95.72.033-.236.05-.476.05-.72 0-2.76-2.24-5-5-5S1 3.24 1 6c0 .244.017.484.05.72C1.577 6.27 2.257 6 3 6v5.6zM11 0v2h7V0h-7zm2 3v2h5V3h-5zm1 3v2h4V6h-4z"/></g></g></svg>
```
Comment (robwalch) [inline assets/fonts/svgs/audio-tracks.svg]: Move to  assets/fonts/svg/*

File: src/js/jwplayer.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 define([
     'api/global-api',
-    '../css/styles.less'
+    '../css/styles.less',
```
Comment (robwalch) [inline src/js/jwplayer.js]: rename styles (error screen?) and include in jwplayer.less

File: test/ui-test.html [hunk 5]
```diff
@@ -0,0 +1,157 @@
+<!DOCTYPE html>
```
Comment (robwalch) [inline test/ui-test.html]: move test/css-skins/

File: src/css/jwplayer.less [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+.jwplayer .jw-controls {
+    pointer-events: none;
+}
+
+.jwplayer.jw-user-inactive .jw-controls {
+    pointer-events: all;
+}
+
+.jwplayer .jw-controls .jw-dock-buttons, .jwplayer .jw-controls .jw-controlbar, .jwplayer .jw-controls .jw-skip, .jwplayer .jw-controls .jw-display-icon, .jwplayer .jw-controls .jw-preview, .jwplayer .jw-controls .jw-logo {
```
Comment (DonatoB) [inline src/css/jwplayer.less]: I would use less nested rules here

    .jwplayer .jw-controls {
        .jw-controlbar, .jw-skip, ...

File: src/css/jwplayer.less [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    font-style: normal;
+    font-weight: normal;
+    line-height: 1em;
+    color: @font-color;
+    text-align: center;
+    font-variant: normal;
+    font-stretch: normal;
+}
+
+@import "imports/display.less";
```
Comment (DonatoB) [inline src/css/jwplayer.less]: why aren't these at the top?

File: src/css/jwplayer.less [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+    bottom: 0;
+    left: 0;
+    display: none;
+}
+
+.jw-aspect {
+    display: none;
+}
+
+.jwplayer.aspect-mode {
```
Comment (DonatoB) [inline src/css/jwplayer.less]: Everywhere else, the classes are prefixed with jw-, we should continue that here.

File: src/css/imports/vars.less [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+@active-color: #eee;
+@inactive-color: #aaa;
+@buffer-color: #202020;
+@hover-color: @active-color;
+@thumb-color: #999;
+
+@text-transform: normal;
+@font-weight: normal;
+@font-size: 0.75em;
+@max-width: 600px;
```
Comment (DonatoB) [inline src/css/imports/vars.less]: This variable name doesn't tell us that this is the controlbars width.

File: test/css-skins/ui-test.html [hunk 10]
```diff
@@ -0,0 +1,157 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>UI Test</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
+    <script type="text/javascript" src="../../bin-debug/jwplayer.js"></script>
+</head>
+<body style="width: 100%; height: 2000px">
+    <div class="jwplayer playlist-none jw-user-inactive" tabindex="0" style="width: 854px; height: 480px; opacity: 1; background-color: rgb(0, 0, 0);">
```
Comment (DonatoB) [inline test/css-skins/ui-test.html]: Can "playlist-none" be removed?

Which of these inline styles can also be removed?

File: test/css-skins/ui-test.html [hunk 11]
```diff
@@ -0,0 +1,157 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>UI Test</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
+    <script type="text/javascript" src="../../bin-debug/jwplayer.js"></script>
+</head>
+<body style="width: 100%; height: 2000px">
+    <div class="jwplayer jw-user-inactive" tabindex="0" style="width: 854px; height: 480px; opacity: 1; background-color: rgb(0, 0, 0);">
+        <div class="jw-click" style="display: none;">
```
Comment (DonatoB) [inline test/css-skins/ui-test.html]: jw-click is a little ambiguous, perhaps .jw-context-menu is more clear.

File: test/css-skins/ui-test.html [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+                                    </div>
+                                </div>
+                            </span>
+                        </span>
+                    </span>
+                    <span class="jw-group jw-right">
+                        <span class="jw-duration">
+                            <span class="jw-text">14:48</span>
+                        </span>
+                        <span class="jw-icon inline-icon tooltip-icon jw-icon-hd-on" style="display: inline-block;">
```
Comment (DonatoB) [inline test/css-skins/ui-test.html]: inline-icon and tooltip-icon should have jw- prefix

**Comment by robwalch** [comment]
@jw-kaurand pleas fix merge conflict

**Comment by jw-kaurand** [comment]
on this. conflict fixed and testing it again

**Comment by robwalch** [comment]
thanks!

**Comment by robwalch** [comment]
@jw-kaurand add corresponding package.json and Gruntfile.js changes to commercial.

**Comment by jw-kaurand** [comment]
To be merged at the same time as with https://github.com/jwplayer/jwplayer-commercial/pull/504

**Pull request merged**